wire-ios-sync-engine/Source/Use cases/SnoozeCertificateEnrollmentUseCase.swift

```diff
@@ -0,0 +1,123 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+// sourcery: AutoMockable
+public protocol SnoozeCertificateEnrollmentUseCaseProtocol {
+    func start() async
+    func stop()
+}
+
+final class SnoozeCertificateEnrollmentUseCase: SnoozeCertificateEnrollmentUseCaseProtocol {
+
+    // MARK: - Properties
+
+    private let e2eiFeature: Feature.E2EI
+    private let gracePeriodRepository: GracePeriodRepository
+    private let recurringActionService: RecurringActionServiceInterface
+    private let selfClientCertificateProvider: SelfClientCertificateProviderProtocol
+    private let actionId: String
+
+    // MARK: - Life cycle
+
+    init(e2eiFeature: Feature.E2EI,
+         gracePeriodRepository: GracePeriodRepository,
+         recurringActionService: RecurringActionServiceInterface,
+         selfClientCertificateProvider: SelfClientCertificateProviderProtocol,
+         accountId: UUID) {
+        self.e2eiFeature = e2eiFeature
+        self.gracePeriodRepository = gracePeriodRepository
+        self.recurringActionService = recurringActionService
+        self.selfClientCertificateProvider = selfClientCertificateProvider
+        self.actionId = "\(accountId).enrollCertificate"
+    }
+
+    // MARK: - Methods
+
+    func start() async {
```

Comment:
As it is a useCase can we keep the naming invoke() method?

Reply:
https://github.com/wireapp/wire-ios/pull/1029